include/lsst/meas/astrom/SipTransform.h

```diff
@@ -0,0 +1,357 @@
+     *  output scalings (respectively).
+     */
+    static SipForwardTransform convert(ScaledPolynomialTransform const & scaled);
+
+    /**
+     *  Construct a SipForwardTransform from its components.
+     *
+     *  @param[in]   pixelOrigin    CRPIX @f$(u_0,v_0)@f$
+     *  @param[in]   cdMatrix       CD matrix @f$Z@f$
+     *  @param[in]   forwardSipPoly Polynomial transform @f$(A,B)@f$
+     */
+    SipForwardTransform(
+        afw::geom::Point2D const & pixelOrigin,
+        afw::geom::LinearTransform const & cdMatrix,
+        PolynomialTransform const & forwardSipPoly
+    ) :
+        SipTransformBase(pixelOrigin, cdMatrix, forwardSipPoly)
+    {}
+
+    SipForwardTransform(SipForwardTransform const & other) = default;
+
+    SipForwardTransform(SipForwardTransform && other) = default;
+
+    SipForwardTransform & operator=(SipForwardTransform const & other) = default;
+
+    SipForwardTransform & operator=(SipForwardTransform && other) = default;
+
+    void swap(SipForwardTransform & other) {
+        SipTransformBase::swap(other);
+    }
+
+    /**
+     * Return an approximate affine transform at the given point.
+     */
+    afw::geom::AffineTransform linearize(afw::geom::Point2D const & in) const;
+
+    /**
+     * Apply the transform to a point.
+     */
+    afw::geom::Point2D operator()(afw::geom::Point2D const & uv) const;
+
+};
+
+
+/**
+ *  A transform that maps intermediate world coordinates to pixel coordinates
+ *  according to the SIP convention.
+ *
+ *  The SIP reverse transform is defined as
+ *  @f[
+ *     \left[\begin{array}{ c }
+ *       u \\
+ *       v
+ *     \end{array}\right]
+ *     = \left[\begin{array}{ c }
+ *       u_0 + U + {\displaystyle\sum_{p,q}^{0 \le p + q \le N}} \mathrm{AP}_{p,q} U^p V^q \\
+ *       v_0 + V + {\displaystyle\sum_{p,q}^{0 \le p + q \le N}} \mathrm{BP}_{p,q} U^p V^q \\
+ *     \end{array}\right]
+ *  @f]
+ *  with
+ *  @f[
+ *     \left[\begin{array}{ c }
+ *       U \\
+ *       V
+ *     \end{array}\right]
+ *     = \mathbf{Z}^{-1}
+ *     \left[\begin{array}{ c }
+ *       x \\
+ *       y
+ *     \end{array}\right]
+ *  @f]
+ *  and
+ *   - @f$(u,v)@f$ are pixel coordinates.
+ *   - @f$(x,y)@f$ are "intermediate world coordinates" -- the result of
+ *     applying the gnomonic (TAN) projection at sky origin CRVAL to sky
+ *     coordinates).
+ *   - @f$\mathbf{Z}@f$ is the @f$2 \times 2@f$ linear transform (@f$\mathrm{CD}@f$) matrix.
+ *   - @f$(u_0,v_0)@f$ is the pixel origin @f$\mathrm{CRVAL}@f$.
+ *   - @f$\mathrm{AP}@f$, @f$\mathrm{BP}@f$ are the polynomial coefficients of
+ *     the reverse transform.
+ *
+ */
+class SipReverseTransform : public SipTransformBase {
+public:
+
+    /**
+     *  Convert a PolynomialTransform to an equivalent SipReverseTransform.
+     *
+     *  @param[in]   poly           PolynomialTransform to convert.
+     *  @param[in]   pixelOrigin    CRPIX @f$(u_0,v_0)@f$
```

Comment:
Again, anywhere you mention CRPIX please be explicit about the zero convention (that's safer than putting it in one place and hoping folks stumble across it).